### PR TITLE
Make Moreau the default and support direct cones

### PR DESCRIPTION
**Pull request by PTNobel**
ChatGPT generated:

Select Moreau by default for PyTorch and JAX, install it as a core dependency, and update the usage and migration documentation. Explicit `solver=cp.DIFFCP` retains the previous backend; MLX continues to default to DIFFCP.

Forward CVXPY's direct-cone metadata and support nonnegative, SOC, exponential, power, generalized-power, and PSD cones. Recover direct-cone duals with gradients, apply PSD coordinate transforms consistently to problem data and outputs, preserve native warm-start values, and handle pure direct LPs with no P or A matrix. Avoid modifying CVXPY's cached parametrization when constructing multiple layers.

Preserve the SciPy CPU sparse-matmul path behind an explicit compilation boundary and normalize fixed matrix dimensions to Python integers. Strengthen CPU/CUDA compilation tests to require actual Inductor work without compiler errors, fullgraph compilation of tensor preprocessing, and correct primal/dual values and gradients.

This is the foundation of the Moreau/MLX stack, followed by #249. It requires current CVXPY and the Moreau fixes in [moreau-project/moreau#13](https://github.com/moreau-project/moreau/pull/13) and [moreau-project/moreau#22](https://github.com/moreau-project/moreau/pull/22). Both native PRs are merged; this remains a draft until the required Moreau changes are available to the package-based CI installation. Full-layer `fullgraph=True` is not promised: native solver calls remain explicit graph breaks.

The combined branch passes 22 CPU checks with PyTorch 2.14 and 22 CUDA checks with PyTorch 2.9.1+cu128 on an NVIDIA T500, with zero skips. Each run includes seven strict compile cases and fifteen default-backend/direct-cone/PSD integration cases across Torch and JAX. These checks use CVXPY `773163ebd`, the Moreau #22 wrapper, and native CPU/CUDA builds from `4b1b73627fff8ceaec002198b3ada080ebc28537`.

### CODE REVIEW

File: docs/guide/solvers.md [hunk 1]
```diff
@@ -211,12 +225,57 @@ b_jax = jax.random.normal(jax.random.PRNGKey(1), shape=(m,))
...
+solution and reuses it when the batch size matches:
 
 ```python
 (x_sol,) = layer(A_tch, b_tch, warm_start=True)
 ```
 
+### Direct Cones and Semidefinite Programs
+
+With a CVXPY version that supports direct-cone extraction (currently CVXPY
+master), Moreau can impose eligible nonnegative, second-order, exponential,
```
Comment (PTNobel) [inline docs/guide/solvers.md]: This needs an update
